pkg/util/cluster/aad.go

```diff
@@ -61,6 +61,23 @@ func (c *Cluster) createServicePrincipal(ctx context.Context, appID string) (str
 }
 
 func (c *Cluster) deleteApplication(ctx context.Context, appID string) error {
+
```

Comment:
let's not add this functionality to deleteApplication: create a new function (in cluster.go) and call it before calling deleteApplication.